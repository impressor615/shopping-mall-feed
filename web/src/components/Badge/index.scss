@import "~@/assets/styles/variables";

.badge {
  display: inline-block;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  padding: 10px;

  color: $darkgrey;
  font-weight: bold;
  font-size: 25px;

  text-align: center;

  &-gold, &-silver, &-bronze {
    color: $white;
  }

  &-gold {
    background: $gold;
  }

  &-silver {
    background: $silver;
  }

  &-bronze {
    background: $bronze;
  }
}