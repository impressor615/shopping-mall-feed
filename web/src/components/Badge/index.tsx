import "./index.scss";

import classnames from "classnames";
import React from "react";

interface BadgeProps {
  rank: number;
}

const Badge = ({ rank }: BadgeProps) => (
  <span className={classnames(
    "badge",
    { "badge-gold": rank === 1 },
    { "badge-silver": rank === 2 },
    { "badge-bronze": rank === 3 },
  )}>
    { rank }
  </span>
);

export default Badge;
