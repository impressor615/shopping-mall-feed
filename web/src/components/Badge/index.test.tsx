import enzyme from "enzyme";
import React from "react";

import Badge from "@/components/Badge";

describe("Badge", () => {
  it("should render gold badge", () => {
    const rank = 1;
    const wrapper = enzyme.shallow(<Badge rank={rank} />);
    expect(wrapper.text()).toEqual(rank.toString());
    expect(wrapper.find(".badge-gold")).toHaveLength(1);
  });

  it("should render silver badge", () => {
    const rank = 2;
    const wrapper = enzyme.shallow(<Badge rank={rank} />);
    expect(wrapper.text()).toEqual(rank.toString());
    expect(wrapper.find(".badge-silver")).toHaveLength(1);
  });

  it("should render bronze badge", () => {
    const rank = 3;
    const wrapper = enzyme.shallow(<Badge rank={rank} />);
    expect(wrapper.text()).toEqual(rank.toString());
    expect(wrapper.find(".badge-bronze")).toHaveLength(1);
  });

  it("should render normal rank", () => {
    const rank = 4;
    const wrapper = enzyme.shallow(<Badge rank={rank} />);
    expect(wrapper.text()).toEqual(rank.toString());
    expect(wrapper.find(".badge-gold")).toHaveLength(0);
    expect(wrapper.find(".badge-silver")).toHaveLength(0);
    expect(wrapper.find(".badge-bronze")).toHaveLength(0);
  });

  it("should be matched with snapshot", () => {
    const wrapper = enzyme.shallow(<Badge rank={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});
