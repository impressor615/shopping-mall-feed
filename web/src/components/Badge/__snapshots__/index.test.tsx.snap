// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should be matched with snapshot 1`] = `
<span
  className="badge badge-gold"
>
  1
</span>
`;
